contracts/upgradeable_contracts/BasicOmnibridge.sol

```diff
@@ -37,6 +37,20 @@ abstract contract BasicOmnibridge is
     using SafeERC20 for IERC677;
     using SafeMath for uint256;
 
+    // Workaround for storing variable up-to-32 bytes suffix
+    uint256 private immutable SUFFIX_SIZE;
+    bytes32 private immutable SUFFIX;
+
+    // Since contract is intended to be deployed under EternalStorageProxy, only constant and immutable variables can be set here
+    constructor(string memory _suffix) {
```

Comment:
I have no ability to initialize the storage on the contract deployment stage since the proxy pattern is used. Consider to add this logic in `initialize` method.

Reply:
Correct. However, since immutable variables cannot be changed anywhere outside from constructor, they are being hardcoded into the run-time bytecode. This makes them also accessible under the proxy pattern. This also allows to not use expensive SLOAD operation for reading this suffix.

Reply:
thanks! did not know about "immutable".